src/main/java/frc/robot/commands/dropping/AutoAlignCommand.java

```diff
@@ -0,0 +1,239 @@
+
+    private final InstantCommand[] setTargetCommands;
+    private final HashMap<PlacePosition, GenericEntry> booleanEntries;
+
+    private volatile PlacePosition targetPlacePosition;
+    private volatile AlignToNodeCommand wrappedCommand;
+
+    private final ShuffleboardTab shuffleboardTab;
+
+    /**
+     * Creates an auto-align command from a given swerve subsystem, elevator subsystem, and
+     * whether the robot is currently on the red team. This command is an `InstantCommand` that
+     * schedules an `AlignToNodeCommand` corresponding to the closest node to the current robot
+     * position.
+     * 
+     * @param swerveSubsystem The swerve subsystem.
+     * @param tiltedElevatorSubsystem The tilted elevator subsystem.
+     * @param isRed Whether the robot is on the red team.
+     */
+    public AutoAlignCommand(
+        BaseSwerveSubsystem swerveSubsystem, TiltedElevatorSubsystem tiltedElevatorSubsystem,
+        boolean isRed
+    ) {
+        this.swerveSubsystem = swerveSubsystem;
+        this.tiltedElevatorSubsystem = tiltedElevatorSubsystem;
+        this.isRed = isRed;
+
+        addRequirements(swerveSubsystem, tiltedElevatorSubsystem);
+
+        // Initialize shuffleboard alignment buttons
+        shuffleboardTab = Shuffleboard.getTab("Grid");
+        booleanEntries = new HashMap<>();
+
+        PlacePosition[] positions = PlacePosition.values();
+        setTargetCommands = new InstantCommand[positions.length];
+
+        for (int i = 0; i < positions.length; i++) {
+            PlacePosition position = positions[i];
+
+            // When pressed, set the target place position to the given value and align to the node.
+            InstantCommand command = new InstantCommand(() -> scheduleAlignCommandWith(position), swerveSubsystem, tiltedElevatorSubsystem);
+            setTargetCommands[i] = command;
+
+            // Add command and boolean indicator to grid
+            int column = getShuffleboardColumn(position);
+            int row = getShuffleboardRow(position);
+
+            shuffleboardTab.add(position.name(), command).withPosition(column, row + 1);
+            booleanEntries.put(
+                position,
+                shuffleboardTab.add("T: " + position.name(), false).withPosition(column, row).getEntry()
+            );
+        }
+    }
+
+    @Override
+    public void initialize() {
+        Pose2d initialPose = swerveSubsystem.getRobotPosition();
+        ElevatorState currentElevatorState = tiltedElevatorSubsystem.getState();
+
+        // Align with closest place position
+        scheduleAlignCommandWith(getClosestPlacePosition(initialPose, currentElevatorState, isRed));
+    }
+
+    /**
+     * Aligns to the node immediately left of the currently selected node (from the perspective of the drivers).
+     * This is a no-op if there isn't yet a node selected, or if the selected node is the left-most node already (C3).
+     */
+    public void alignLeft() {
```

Comment:
"left" and "right" are probably misleading now. flipping based on alliance color doesn't seem to be implemented yet, either